pkg/controller/cloud/node_controller.go

```diff
@@ -62,17 +62,14 @@ func NewCloudNodeController(
 	nodeInformer coreinformers.NodeInformer,
 	kubeClient clientset.Interface,
 	cloud cloudprovider.Interface,
-	nodeStatusUpdateFrequency time.Duration) *CloudNodeController {
+	nodeStatusUpdateFrequency time.Duration) (*CloudNodeController, error) {
 
 	eventBroadcaster := record.NewBroadcaster()
 	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cloud-node-controller"})
 	eventBroadcaster.StartLogging(klog.Infof)
-	if kubeClient != nil {
```

Comment:
Let's add a cloud.Instances() check similar to `NewCloudNodeLifecycleController`?

Reply:
It sounds good.

Reply:
`cloud.Instances()` has been added.